packages/rocketchat-search/server/service/validationService.js

```diff
@@ -0,0 +1,51 @@
+import _ from 'underscore';
+
+import SearchLogger from '../logger/logger';
+
+class ValidationService {
+	constructor() {}
+
+	validateSearchResult(result) {
+
+		const uid = Meteor.userId();
+		//get subscription for message
+		if (result.message) {
+			result.message.docs = _.chain(result.message.docs)
+				.each((msg) => {
+					const subscription = Meteor.call('canAccessRoom', msg.rid, uid);
+					if (subscription) {
+						msg.r = {name: subscription.name, t: subscription.t};
+						msg.username = subscription.username;
+						msg.valid = true;
+						SearchLogger.debug(`user ${ uid } can access ${ msg.rid } ( ${ subscription.t === 'd' ? subscription.username : subscription.name } )`);
+					} else {
+						SearchLogger.debug(`user ${ uid } can NOT access ${ msg.rid }`);
+					}
+				})
+				.filter((msg) => {
+					SearchLogger.debug(JSON.stringify(msg, null, 2));
```

Comment:
some considerations here:

- stringifying every message even if the log is not shown (due to higher debug level) seems we're wasting CPU for no return
- not sure about showing every message's content on server logs